From 2de221ce94b657f9c9a75aa3cc0cb84dbb5da78b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Olivier=20Cr=C3=AAte?= <olivier.crete@collabora.co.uk>
Date: Mon, 28 Jan 2013 20:28:09 +0100
Subject: [PATCH] gtk-doc*.mak: Put the API version in the html book name

This causes devhelp2 files to get a version. Due to that devhelp can show it
along with the 0.10 version.
---
 gtk-doc-plugins.mak | 11 ++++-------
 gtk-doc.mak         | 11 ++++-------
 2 files changed, 8 insertions(+), 14 deletions(-)

diff -urNp -x '*.orig' gst-plugins-bad-0.10.23.org/common/gtk-doc-plugins.mak gst-plugins-bad-0.10.23/common/gtk-doc-plugins.mak
--- gst-plugins-bad-0.10.23.org/common/gtk-doc-plugins.mak	2011-12-30 19:32:24.000000000 +0100
+++ gst-plugins-bad-0.10.23/common/gtk-doc-plugins.mak	2021-10-06 23:28:06.340365779 +0200
@@ -162,19 +162,10 @@ scan-build.stamp: $(HFILE_GLOB) $(EXTRA_
 	    --ignore-headers="$(IGNORE_HFILES)";			\
 	touch scan-build.stamp
 
-#### update templates; done on every build ####
+#### generate templates; done on every build ####
 
-# in a non-srcdir build, we need to copy files from the previous step
-# and the files from previous runs of this step
-tmpl-build.stamp: $(DOC_MODULE)-decl.txt $(SCANOBJ_FILES) $(DOC_MODULE)-sections.txt $(DOC_OVERRIDES)
-	@echo '  DOC   Rebuilding template files'
-	@if test x"$(srcdir)" != x. ; then				\
-	    for f in $(SCANOBJ_FILES) $(SCAN_FILES);			\
-	    do								\
-	        if test -e $(srcdir)/$$f; then cp -u $(srcdir)/$$f . ; fi;	\
-	    done;							\
-	fi
-	@gtkdoc-mktmpl --module=$(DOC_MODULE)
+tmpl-build.stamp:
+	@echo '  DOC   Building template files'
 	@$(PYTHON) \
 		$(top_srcdir)/common/mangle-tmpl.py $(srcdir)/$(INSPECT_DIR) tmpl
 	@touch tmpl-build.stamp
@@ -192,9 +183,13 @@ sgml-build.stamp: tmpl.stamp scan-build.
 		$(top_srcdir)/common/plugins.xsl $$a > xml/`basename $$a`; done
 	@for f in $(EXAMPLE_CFILES); do \
 		$(PYTHON) $(top_srcdir)/common/c-to-xml.py $$f > xml/element-`basename $$f .c`.xml; done
-	@gtkdoc-mkdb \
+	@_source_dir='' ;						\
+	for i in $(DOC_SOURCE_DIR) ; do					\
+	    _source_dir="$${_source_dir} --source-dir=$$i" ;	        \
+	done ;								\
+	gtkdoc-mkdb \
 		--module=$(DOC_MODULE) \
-		--source-dir=$(DOC_SOURCE_DIR) \
+		$${_source_dir} \
 		 --expand-content-files="$(expand_content_files)" \
 		--main-sgml-file=$(srcdir)/$(DOC_MAIN_SGML_FILE) \
 		--output-format=xml \
@@ -223,10 +218,7 @@ html-build.stamp: sgml.stamp $(DOC_MAIN_
 	    mkhtml_options="$$mkhtml_options --verbose"; \
 	  fi; \
 	fi; \
-	cd html && gtkdoc-mkhtml $$mkhtml_options $(DOC_MODULE) $(DOC_MAIN_SGML_FILE)
-	@mv html/index.sgml html/index.sgml.bak
-	@$(SED) "s/ href=\"$(DOC_MODULE)\// href=\"$(DOC_MODULE)-@GST_MAJORMINOR@\//g" html/index.sgml.bak >html/index.sgml
-	@rm -f html/index.sgml.bak
+	cd html && gtkdoc-mkhtml $$mkhtml_options $(DOC_MODULE)-@GST_MAJORMINOR@ $(DOC_MAIN_SGML_FILE)
 	@rm -f html/$(DOC_MAIN_SGML_FILE)
 	@rm -rf html/xml
 	@rm -f html/version.entities
@@ -290,9 +282,9 @@ install-data-local:
 	      $(INSTALL_DATA) $$i $(DESTDIR)$(TARGET_DIR); \
 	    done; \
 	  fi; \
-	  echo '-- Installing $(builddir)/html/$(DOC_MODULE).devhelp2' ; \
-	  if test -e $(builddir)/html/$(DOC_MODULE).devhelp2; then \
-	            $(INSTALL_DATA) $(builddir)/html/$(DOC_MODULE).devhelp2 \
+	  echo '-- Installing $(builddir)/html/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2' ; \
+	  if test -e $(builddir)/html/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2; then \
+	            $(INSTALL_DATA) $(builddir)/html/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2 \
 	            $(DESTDIR)$(TARGET_DIR)/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2; \
 	  fi; \
 	  (which gtkdoc-rebase >/dev/null && \
diff -urNp -x '*.orig' gst-plugins-bad-0.10.23.org/common/gtk-doc.mak gst-plugins-bad-0.10.23/common/gtk-doc.mak
--- gst-plugins-bad-0.10.23.org/common/gtk-doc.mak	2011-12-30 19:32:24.000000000 +0100
+++ gst-plugins-bad-0.10.23/common/gtk-doc.mak	2021-10-06 23:28:06.340365779 +0200
@@ -104,7 +104,11 @@ $(DOC_MODULE)-decl.txt $(SCANOBJ_FILES)
 
 sgml-build.stamp: setup-build.stamp $(DOC_MODULE)-decl.txt $(SCANOBJ_FILES) $(DOC_MODULE)-sections.txt $(expand_content_files)
 	@echo '  DOC   Building XML'
-	@gtkdoc-mkdb --module=$(DOC_MODULE) --source-dir=$(DOC_SOURCE_DIR)  --expand-content-files="$(expand_content_files)" --main-sgml-file=$(DOC_MAIN_SGML_FILE) --output-format=xml $(MKDB_OPTIONS)
+	@_source_dir='' ;						\
+	for i in $(DOC_SOURCE_DIR) ; do					\
+	    _source_dir="$${_source_dir} --source-dir=$$i" ;	        \
+	done ;							        \
+	gtkdoc-mkdb --module=$(DOC_MODULE) $${_source_dir}  --expand-content-files="$(expand_content_files)" --main-sgml-file=$(DOC_MAIN_SGML_FILE) --output-format=xml $(MKDB_OPTIONS)
 	@cp ../version.entities xml
 	@touch sgml-build.stamp
 
@@ -130,10 +134,7 @@ html-build.stamp: sgml.stamp $(DOC_MAIN_
 	if test "$(?)" = "0"; then \
 	  mkhtml_options=--path="$(abs_srcdir)"; \
 	fi; \
-	cd html && gtkdoc-mkhtml $$mkhtml_options $(MKHTML_OPTIONS) $(DOC_MODULE) ../$(DOC_MAIN_SGML_FILE)
-	@mv html/index.sgml html/index.sgml.bak
-	@$(SED) "s/ href=\"$(DOC_MODULE)\// href=\"$(DOC_MODULE)-@GST_MAJORMINOR@\//g" html/index.sgml.bak >html/index.sgml
-	@rm -f html/index.sgml.bak
+	cd html && gtkdoc-mkhtml $$mkhtml_options $(MKHTML_OPTIONS) $(DOC_MODULE)-@GST_MAJORMINOR@ ../$(DOC_MAIN_SGML_FILE)
 	@rm -rf html/xml
 	@rm -f version.entities
 	@test "x$(HTML_IMAGES)" = "x" ||  ( cd $(srcdir) && cp $(HTML_IMAGES) $(abs_builddir)/html )
@@ -189,9 +190,9 @@ install-data-local:
 	    echo '-- Installing '$$i ; \
 	    $(INSTALL_DATA) $$i $(DESTDIR)$(TARGET_DIR); \
 	  done; \
-	  echo '-- Installing $(builddir)/html/$(DOC_MODULE).devhelp2' ; \
-	  if test -e $(builddir)/html/$(DOC_MODULE).devhelp2; then \
-	            $(INSTALL_DATA) $(builddir)/html/$(DOC_MODULE).devhelp2 \
+	  echo '-- Installing $(builddir)/html/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2' ; \
+	  if test -e $(builddir)/html/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2; then \
+	            $(INSTALL_DATA) $(builddir)/html/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2 \
 	            $(DESTDIR)$(TARGET_DIR)/$(DOC_MODULE)-@GST_MAJORMINOR@.devhelp2; \
 	  fi; \
 	  (which gtkdoc-rebase >/dev/null && \
diff -urNp -x '*.orig' gst-plugins-bad-0.10.23.org/common/mangle-tmpl.py gst-plugins-bad-0.10.23/common/mangle-tmpl.py
--- gst-plugins-bad-0.10.23.org/common/mangle-tmpl.py	2011-12-11 20:19:22.000000000 +0100
+++ gst-plugins-bad-0.10.23/common/mangle-tmpl.py	2021-10-06 23:28:06.340365779 +0200
@@ -2,21 +2,12 @@
 # vi:si:et:sw=4:sts=4:ts=4
 
 """
-use the output from gst-xmlinspect.py to mangle tmpl/*.sgml and
-insert/overwrite Short Description and Long Description
+use the files from inspect/*.xml to create mininal tmpl/*.sgml files containing
+'Short Description' and 'Long Description' to inject element details into the
+docbook files produced by gtkdoc-mkdb
 """
 
-# FIXME: right now it uses pygst and scans on its own;
-# we really should use inspect/*.xml instead since the result of
-# gst-xmlinspect.py is committed by the docs maintainer, who can be
-# expected to have pygst, but this step should be done for every docs build,
-# so no pygst allowed
-
-# read in inspect/*.xml
-# for every tmpl/element-(name).xml: mangle with details from element
-
 import glob
-import re
 import sys
 import os
 
@@ -26,37 +17,12 @@ class Tmpl:
         self._sectionids = []
         self._sections = {}
 
-    def read(self):
-        """
-        Read and parse the sections from the given file.
-        """
-        lines = open(self.filename).readlines()
-        matcher = re.compile("<!-- ##### SECTION (\S+) ##### -->\n")
-        id = None
-
-        for line in lines:
-            match = matcher.search(line)
-            if match:
-                id = match.expand("\\1")
-                self._sectionids.append(id)
-                self._sections[id] = []
-            else:
-                if not id:
-                    sys.stderr.write(
-                        "WARNING: line before a SECTION header: %s" % line)
-                else:
-                    self._sections[id].append(line)
-
-    def get_section(self, id):
-        """
-        Get the content from the given section.
-        """
-        return self._sections[id]
-
     def set_section(self, id, content):
         """
         Replace the given section id with the given content.
         """
+        if not id in self._sectionids:
+            self._sectionids.append(id)
         self._sections[id] = content
 
     def output(self):
@@ -71,14 +37,10 @@ class Tmpl:
 
         return "".join(lines)
 
-    def write(self, backup=False):
+    def write(self):
         """
         Write out the template file again, backing up the previous one.
         """
-        if backup:
-            target = self.filename + ".mangle.bak"
-            os.rename(self.filename, target)
-
         handle = open(self.filename, "w")
         handle.write(self.output())
         handle.close()
@@ -134,30 +96,28 @@ def main():
     inspectdir = sys.argv[1]
     tmpldir = sys.argv[2]
 
+    if not os.path.exists (tmpldir):
+        os.mkdir(tmpldir)
+
     # parse all .xml files; build map of element name -> short desc
     #for file in glob.glob("inspect/plugin-*.xml"):
     elements = {}
     for file in glob.glob("%s/plugin-*.xml" % inspectdir):
         elements.update(get_elements(file))
 
-    for file in glob.glob("%s/element-*.sgml" % tmpldir):
-        base = os.path.basename(file)
-        element = base[len("element-"):-len(".sgml")]
+    for element in elements.keys():
+        file = "%s/element-%s.sgml" % (tmpldir, element)
         tmpl = Tmpl(file)
-        tmpl.read()
-        if element in elements.keys():
-            description = elements[element]['description']
-            tmpl.set_section("Short_Description", "%s\n\n" % description)
 
-        # put in an include if not yet there
+        description = elements[element]['description']
+        tmpl.set_section("Short_Description", "%s\n" % description)
+
+        # add include for details
         line = '<include xmlns="http://www.w3.org/2003/XInclude" href="' + \
             'element-' + element + '-details.xml">' + \
             '<fallback xmlns="http://www.w3.org/2003/XInclude" />' + \
             '</include>\n'
-        section = tmpl.get_section("Long_Description")
-        if not section[0]  == line:
-            section.insert(0, line)
-        tmpl.set_section("Long_Description", section)
+        tmpl.set_section("Long_Description", line)
         tmpl.write()
 
 main()
